### PR TITLE
PR: Add "colour.models.rgb.transfer_functions.log" module.

### PR DESCRIPTION
**Pull request by njwardhan**
This pull request adds the `log.py` module to the 'transfer functions' sub-package in colour models, and contains the following functions:
- `log_encoding_Log2`
- `log_decoding_Log2`

### CODE REVIEW

**Comment by KelSolaar** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- colour/models/rgb/transfer_functions/log.py  1
- colour/models/rgb/transfer_functions/tests/test_log.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/colour-science/colour/pullRequest?prid=5693865&bid=18633918)

File: colour/models/rgb/transfer_functions/__init__.py [hunk 1]
```diff
@@ -58,6 +58,7 @@
                         log_encoding_SLog3, log_decoding_SLog3)
 from .srgb import eotf_inverse_sRGB, eotf_sRGB
 from .viper_log import log_encoding_ViperLog, log_decoding_ViperLog
+from .log import log_encoding_Log2, log_decoding_Log2
```
Comment (KelSolaar) [inline colour/models/rgb/transfer_functions/__init__.py:61]: Given the module does not depend on anything nor anything really depends on it, please import it in alphabetical order.

File: colour/models/rgb/transfer_functions/__init__.py [hunk 2]
```diff
@@ -116,6 +117,7 @@
 ]
 __all__ += ['eotf_inverse_sRGB', 'eotf_sRGB']
 __all__ += ['log_encoding_ViperLog', 'log_decoding_ViperLog']
+__all__ += ['log_encoding_Log2', 'log_decoding_Log2']
```
Comment (KelSolaar) [inline colour/models/rgb/transfer_functions/__init__.py:120]: Alphabetical ordering here too please.

File: colour/models/rgb/transfer_functions/__init__.py [hunk 3]
```diff
@@ -142,7 +144,8 @@
     'S-Log3': log_encoding_SLog3,
     'T-Log': log_encoding_FilmLightTLog,
     'V-Log': log_encoding_VLog,
-    'ViperLog': log_encoding_ViperLog
+    'ViperLog': log_encoding_ViperLog,
+    'Log2': log_encoding_Log2
```
Comment (KelSolaar) [inline colour/models/rgb/transfer_functions/__init__.py:148]: Likewise :)

File: colour/models/rgb/transfer_functions/__init__.py [hunk 4]
```diff
@@ -152,7 +155,7 @@
     'Canon Log 3', 'Canon Log', 'Cineon', 'D-Log', 'ERIMM RGB', 'F-Log',
     'Filmic Pro 6', 'Log3G10', 'Log3G12', 'Panalog', 'PLog', 'Protune',
     'REDLog', 'REDLogFilm', 'S-Log', 'S-Log2', 'S-Log3', 'T-Log', 'V-Log',
-    'ViperLog'}**
+    'ViperLog', 'Log2'}**
```
Comment (KelSolaar) [inline colour/models/rgb/transfer_functions/__init__.py:158]: Same ;)

File: colour/models/rgb/transfer_functions/__init__.py [hunk 5]
```diff
@@ -170,7 +173,7 @@ def log_encoding(value, function='Cineon', **kwargs):
         'Canon Log 3', 'Canon Log', 'Cineon', 'D-Log', 'ERIMM RGB', 'F-Log',
         'Filmic Pro 6', 'Log3G10', 'Log3G12', 'Panalog', 'PLog', 'Protune',
         'REDLog', 'REDLogFilm', 'S-Log', 'S-Log2', 'S-Log3', 'T-Log',
-        'V-Log', 'ViperLog'}**,
+        'V-Log', 'ViperLog', 'Log2'}**,
```
Comment (KelSolaar) [inline colour/models/rgb/transfer_functions/__init__.py:176]: And same for the remaining instances (even in the docs).

File: colour/models/rgb/transfer_functions/log.py [hunk 6]
```diff
@@ -0,0 +1,142 @@
+#
+"""
+Log2 Shaper Implementation
```
Comment (KelSolaar) [inline colour/models/rgb/transfer_functions/log.py:3]: The title could be _Common Log Encodings_ or alike, what do you think @nick-shaw?

File: colour/models/rgb/transfer_functions/log.py [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+==========================
+
+Defines the *Log2 shaper* log encodings:
```
Comment (KelSolaar) [inline colour/models/rgb/transfer_functions/log.py:6]: _Defines the common log encodings:_

File: colour/models/rgb/transfer_functions/log.py [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+
+__all__ = ['log_encoding_Log2', 'log_decoding_Log2']
+
+
+def log_encoding_Log2(lin,
+                      middle_grey=0.18,
+                      min_exposure=0.18 * 2 ** -6.5,
+                      max_exposure=0.18 * 2 ** 6.5):
+    """
+    Defines the *Log2 shaper* log encoding function.
```
Comment (KelSolaar) [inline colour/models/rgb/transfer_functions/log.py:49]: I think we can drop *shaper* here, the function is generic enough that it is not necessarily used as a shaper.

File: colour/models/rgb/transfer_functions/log.py [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+
+    Parameters
+    ----------
+    lin : numeric or array_like
+          Linear data to undergo encoding.
+    middle_grey : numeric, optional
+                  'Middle Grey' exposure value.
```
Comment (KelSolaar) [inline colour/models/rgb/transfer_functions/log.py:56]: If you want to use quotes, prefer `*`, e.g. `*Middle Grey*`, careful at the tabulation.

File: colour/models/rgb/transfer_functions/log.py [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+    min_exposure : numeric, optional
+                   Minimum exposure level.
+    max_exposure : numeric, optional
+                   Maximum exposure level.
+
+    Returns
+    -------
+    numeric or ndarray
+        Non-linear *Log2 shaper* encoded data
```
Comment (KelSolaar) [inline colour/models/rgb/transfer_functions/log.py:65]: Full stop at the end ;)

File: colour/models/rgb/transfer_functions/log.py [hunk 11]
```diff
@@ -0,0 +1,142 @@
...
+
+    Notes
```
Comment (KelSolaar) [inline colour/models/rgb/transfer_functions/log.py:67]: If the section is empty, we should get rid of it, maybe here is a good place to mention how this function is used to build various shapers in the ACES OCIO configuration.

File: colour/models/rgb/transfer_functions/log.py [hunk 12]
```diff
@@ -0,0 +1,142 @@
...
+    -----
+
+    References
+    ----------
+
```
Comment (KelSolaar) [inline colour/models/rgb/transfer_functions/log.py:72]:
```
:cite:`TheAcademyofMotionPictureArtsandSciencesa`
```

File: colour/models/rgb/transfer_functions/log.py [hunk 13]
```diff
@@ -0,0 +1,142 @@
...
+    -------
+    numeric or ndarray
+        Linear *Log2 shaper* decoded data
+
+    Notes
+    -----
+
+    References
+    ----------
+
```
Comment (KelSolaar) [inline colour/models/rgb/transfer_functions/log.py:122]:
```
:cite:`TheAcademyofMotionPictureArtsandSciencesb`
```

File: colour/models/rgb/transfer_functions/log.py [hunk 14]
```diff
@@ -0,0 +1,142 @@
...
+    Examples
+    --------
+    Linear numeric input gets encoded as follows:
```
Comment (KelSolaar) [inline colour/models/rgb/transfer_functions/log.py:75]: ~gets~ is

File: colour/models/rgb/transfer_functions/tests/test_log.py [hunk 15]
```diff
@@ -0,0 +1,175 @@
...
+__author__ = 'Colour Developers'
+__copyright__ = 'Copyright (C) 2013-2020 - Colour Developers'
+__license__ = 'New BSD License - https://opensource.org/licenses/BSD-3-Clause'
+__maintainer__ = 'Colour Developers'
+__email__ = 'colour-developers@colour-science.org'
+__status__ = 'Production'
+
+__all__ = [
+    'TestLogEncoding_Log2',
+    'TestLogDecoding_Log2',
```
Comment (KelSolaar) [inline colour/models/rgb/transfer_functions/tests/test_log.py:25]: Extra coma at the end prevents Yapf inlining.

File: docs/colour.models.rst [hunk 16]
```diff
@@ -588,6 +588,8 @@ Log Encoding and Decoding
     log_decoding_VLog
     log_encoding_ViperLog
     log_decoding_ViperLog
+    log_encoding_Log2
```
Comment (KelSolaar) [inline docs/colour.models.rst:591]: Alphabetical ordering please :)

File: colour/models/rgb/transfer_functions/__init__.py (hunk 1 above)
Comment (njwardhan) [reply colour/models/rgb/transfer_functions/__init__.py:61]: Noted! Will take care of that in the future! :+1:

File: colour/models/rgb/transfer_functions/__init__.py (hunk 2 above)
Comment (njwardhan) [reply colour/models/rgb/transfer_functions/__init__.py:120]: Noted!

File: colour/models/rgb/transfer_functions/__init__.py (hunk 3 above)
Comment (njwardhan) [reply colour/models/rgb/transfer_functions/__init__.py:148]: Sure :)

File: colour/models/rgb/transfer_functions/__init__.py (hunk 4 above)
Comment (njwardhan) [reply colour/models/rgb/transfer_functions/__init__.py:158]: Noted!

File: colour/models/rgb/transfer_functions/__init__.py (hunk 5 above)
Comment (njwardhan) [reply colour/models/rgb/transfer_functions/__init__.py:176]: Will take care, Thankyou!

File: colour/models/rgb/transfer_functions/log.py (hunk 8 above)
Comment (njwardhan) [reply colour/models/rgb/transfer_functions/log.py:49]: Yeah that would make sense. Will make the change accordingly! :)

File: colour/models/rgb/transfer_functions/log.py (hunk 6 above)
Comment (nick-shaw) [reply colour/models/rgb/transfer_functions/log.py:3]: Would make sense to keep it more general. This module could be expanded later to include things like the generic camera log function for CLF 3.0.

File: colour/models/rgb/transfer_functions/log.py (hunk 10 above)
Comment (njwardhan) [reply colour/models/rgb/transfer_functions/log.py:65]: Noted!

File: colour/models/rgb/transfer_functions/log.py (hunk 12 above)
Comment (njwardhan) [reply colour/models/rgb/transfer_functions/log.py:72]: Thankyou!

File: colour/models/rgb/transfer_functions/log.py (hunk 13 above)
Comment (njwardhan) [reply colour/models/rgb/transfer_functions/log.py:122]: Thankyou!

File: colour/models/rgb/transfer_functions/log.py (hunk 14 above)
Comment (njwardhan) [reply colour/models/rgb/transfer_functions/log.py:75]: Sure :)

File: docs/colour.models.rst (hunk 16 above)
Comment (njwardhan) [reply docs/colour.models.rst:591]: Noted, Thankyou!

File: colour/models/rgb/transfer_functions/log.py (hunk 9 above)
Comment (njwardhan) [reply colour/models/rgb/transfer_functions/log.py:56]: Noted, will take care.

File: colour/models/rgb/transfer_functions/log.py (hunk 11 above)
Comment (njwardhan) [reply colour/models/rgb/transfer_functions/log.py:67]: Sounds good, will go about it!

File: colour/models/rgb/transfer_functions/tests/test_log.py (hunk 15 above)
Comment (njwardhan) [reply colour/models/rgb/transfer_functions/tests/test_log.py:25]: Noted

**Comment by KelSolaar** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- colour/models/rgb/transfer_functions/log.py  1
- colour/models/rgb/transfer_functions/tests/test_log.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/colour-science/colour/pullRequest?prid=5693865&bid=18633918)

File: colour/models/rgb/transfer_functions/log.py [hunk 17]
```diff
@@ -1,9 +1,9 @@
 #
 """
-Log2 Shaper Implementation
+Common Log Encodings
 ==========================
```
Comment (KelSolaar) [inline colour/models/rgb/transfer_functions/log.py:4]: Careful at title marker length: `==========================` should be shorter

File: colour/models/rgb/transfer_functions/log.py [hunk 18]
```diff
@@ -46,38 +46,52 @@ def log_encoding_Log2(lin,
...
+
+    A (48-nits OCIO) shaper having values in a linear
+    domain, can be encoded to a logarithmic domain:
+
+    +-------------------+------------------+
+    | **Shaper Domain** | **Shaper range** |
+    +===================+==================+
+    | [0.002, 16.291]   | [0, 1]           |
+    +-------------------+------------------+
+
```
Comment (KelSolaar) [inline colour/models/rgb/transfer_functions/log.py:82]: Nice! :)

File: colour/models/rgb/transfer_functions/log.py [hunk 19]
```diff
@@ -46,38 +46,52 @@ def log_encoding_Log2(lin,
...
 
+    The common *Log2* encoding function can be used
+    to build linear to logarithmic shapers in the
+    ACES OCIO configuration.
+
+    A (48-nits OCIO) shaper having values in a linear
+    domain, can be encoded to a logarithmic domain:
+
+    +-------------------+------------------+
+    | **Shaper Domain** | **Shaper range** |
```
Comment (KelSolaar) [inline colour/models/rgb/transfer_functions/log.py:78]: Title case _range_ for consistency.

File: colour/models/rgb/transfer_functions/log.py [hunk 20]
```diff
@@ -96,38 +110,52 @@ def log_decoding_Log2(log_norm,
...
 
+    The common *Log2* decoding function can be used
+    to build logarithmic to linear shapers in the
+    ACES OCIO configuration.
+
+    The shaper with logarithmic encoded values can be
+    decoded back to linear domain:
+
+    +-------------------+------------------+
+    | **Shaper Range** | **Shaper Domain** |
```
Comment (KelSolaar) [inline colour/models/rgb/transfer_functions/log.py:142]: Careful, the table will break at the column separator are not aligned, sphinx is very picky with that.

File: colour/models/rgb/transfer_functions/log.py (hunk 17 above)
Comment (njwardhan) [reply colour/models/rgb/transfer_functions/log.py:4]: Ah, should have taken care. Noted anyways!

File: colour/models/rgb/transfer_functions/log.py (hunk 18 above)
Comment (njwardhan) [reply colour/models/rgb/transfer_functions/log.py:82]: Thankyou! :)

File: colour/models/rgb/transfer_functions/log.py (hunk 19 above)
Comment (njwardhan) [reply colour/models/rgb/transfer_functions/log.py:78]: Noted!

File: colour/models/rgb/transfer_functions/log.py (hunk 20 above)
Comment (njwardhan) [reply colour/models/rgb/transfer_functions/log.py:142]: Will take care, Thankyou!

**Comment by KelSolaar** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- colour/models/rgb/transfer_functions/log.py  1
- colour/models/rgb/transfer_functions/tests/test_log.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/colour-science/colour/pullRequest?prid=5693865&bid=18633918)

**Comment by KelSolaar** [comment]
LGTM! Thanks @njwardhan!

**Pull request merged**